redis.conf

```diff
@@ -1419,6 +1419,20 @@ aof-use-rdb-preamble yes
 # the AOF format in a way that may not be compatible with existing AOF parsers.
 aof-timestamp-enabled no
 
+################################ SHUTDOWN #####################################
+
+# Maximum time to wait for replicas when shutting down, in seconds.
+#
+# During shut down, a grace period allows any lagging replicas to catch up with
+# the latest replication offset before the master exists. This period can
+# prevent data loss, especially for deployments without configured disk backups.
+#
+# The 'shutdown-timeout' value is the grace period's duration in seconds. It is
+# only applicable when the instance has replicas. The maximal value is 3600
+# (one hour). To disable the feature, set the value to 0.
```

Comment:
@zuiderkwast Are you sure about having an arbitrary max value limit? In cases where this feature may come in handy, might it make sense to simply wait forever (or until the replica drops, or someone force kills the server)?

Reply:
Good question. I don't have a strong opinion about it. Also, I don't have a use case to wait forever. :-)

One reason to set the maximum relatively low is that it's always possible to increase it in the future, but it won't be possible to decrease it without breaking backward compatibility.

What do you suggest? UINT_MAX? Or should we use a magic value such as -1 to wait forever?

Reply:
let's use INT_MAX (just avoid any possible issues with overflow...
alternatively, we can define "infinite" as one week maybe? (an hour may be too short for some insane users, but a week should do it?)